lib/trino-hive-formats/src/main/java/io/trino/hive/formats/HiveFormatUtils.java

```diff
@@ -237,6 +237,7 @@ private static DecodedTimestamp parseHiveTimestamp(org.joda.time.format.DateTime
 
     public static DecodedTimestamp parseHiveTimestamp(String value)
     {
+        value = value.trim();
```

Comment:
It looks like this method is called from multiple codepaths, some of which have already called `value.trim()` on the argument. Maybe we should add a new private method `parseTrimmedHiveTimestamp` that we can call instead to ensure that we don't introduce more overhead than necessary?